gnovm/pkg/gnolang/go2gno.go

```diff
@@ -137,6 +137,33 @@ func setLoc(fs *token.FileSet, pos token.Pos, n Node) Node {
 	return n
 }
 
+type LocationPlusError struct {
+	pos token.Position
+	msg string
+}
+
+func (ewp *LocationPlusError) Error() string {
+	return fmt.Sprintf("%s: %s", ewp.Location(), ewp.msg)
+}
+
+func (ewp *LocationPlusError) Location() string {
+	if ewp == nil {
+		return "undefined:0"
+	}
+	return fmt.Sprintf("%s:%d:%d", ewp.pos.Filename, ewp.pos.Line, ewp.pos.Column)
+}
+
+func (ewp *LocationPlusError) Message() string {
+	return ewp.msg
+}
+
+func MakeLocationPlusError(pos token.Position, msg string) *LocationPlusError {
+	return &LocationPlusError{
+		msg: msg,
+		pos: pos,
+	}
+}
+
```

Comment:
Would avoid custom type, fmt.Errorf is enough (this is also what we do in `doRecover` in preprocess.go)

Reply:
On the contrary, fmt.Errorf makes us lose valuable information which then causes the regex checks and guessing. But for the sake of maintenance and growth of the library, I'll use fmt.Errorf and so much later once I've made many more improvements I'll draft up a proposal for keeping contextual information. I'll make the update, thanks!